lib/api/api.go

```diff
@@ -362,7 +362,9 @@ func (s *service) Serve(ctx context.Context) error {
 
 	// Wrap everything in basic auth, if user/password is set.
 	if guiCfg.IsAuthEnabled() {
-		handler = basicAuthAndSessionMiddleware("sessionid-"+s.id.String()[:5], guiCfg, s.cfg.LDAP(), handler, s.evLogger)
+		var handlePasswordAuth http.Handler
+		handler, handlePasswordAuth = authAndSessionMiddleware("sessionid-"+s.id.String()[:5], guiCfg, s.cfg.LDAP(), handler, s.evLogger)
+		restMux.Handler(http.MethodPost, "/rest/noauth/authn/password", handlePasswordAuth)
```

Comment:
A little bit of bikeshedding: Is `authn` a generally used shortening for authentication? Otherwise I'd propose just `auth`.

Reply:
My bad - `authc` is a (somewhat) common abbreviation of "authenti**c**ation" (integrity and identity verification), to differentiate it from `authz` for "authori**z**ation" (permissions management). `authn` is just because I'm so used to it from "WebAuthn", which is short for "Web Authentication". So yeah, this should at least be `authc` rather than `authn`. I think `authc` is still worth keeping to differentiate from authorization, but do you have a preference between `authc` and `auth`?

Reply:
I was thinking it must be that, but couldn't find any particularly convincing way to map "c" and "n" uniquely to authorisation and authentication. And with the "z" it wouldn't have helped that that I often use "s" instead of "z". Anyway I don't have a strong opinion on auth vs authc - whatever you prefer.

Reply:
On second thought, there is already the `noauth/` part right next to this, so perhaps it should be `/rest/noauth/auth/password` just to be visually consistent.

Reply:
Fixed in 638c20056123bb7bdc35a519d56bcc6bb08ae83a.